pkg/controller/operators/catalog/operator.go

```diff
@@ -664,7 +666,7 @@ func (o *Operator) ensureResolverSources(logger *logrus.Entry, namespace string)
 		logger.WithField("clientState", client.Conn.GetState()).Debug("source")
 		if client.Conn.GetState() == connectivity.TransientFailure {
 			logger.WithField("clientState", client.Conn.GetState()).Debug("waiting for connection")
-			ctx, _ := context.WithTimeout(context.TODO(), 5*time.Second)
+			ctx, _ := context.WithTimeout(context.TODO(), 2*time.Second)
```

Comment:
I assume that was an arbitrary time out.

Reply:
I was playing with the timeouts - I'm changing it back, thanks for pointing it out. Yeah it's arbitrary. I really hope over the next sprint we can encapsulate connection logic into a separate module (I'm thinking a `ConnectionManager` in a separate thread).